ecs-agent/netlib/platform/common_linux.go

```diff
@@ -520,7 +526,12 @@ func (c *common) generateNetworkConfigFilesForDebugPlatforms(
 		return errors.Wrap(err, "unable to create hostname file for netns")
 	}
 
-	err = c.copyFile(filepath.Join(netNSDir, ResolveConfFileName), "/etc/resolv.conf", taskDNSConfigFileMode)
+	// Check if the /run/netdog/resolv.conf file exists on the host. For Bottlerocket, instead of /etc/resolv.conf,
+	// this is the file we copy to the task network namespace dns config.
+	if _, err = c.os.Stat(srcBRResolv); err == nil {
```

Comment:
Is there a better way to detect if the host is using a bottle rocket AMI?

Reply:
I think we should create a bottle-rocket specific platform in Network builder. Maybe not in this PR, but as a todo long-term. I can take that AI.

Reply:
Agreed. Thanks Jose

Reply:
Is `/run/netdog/resolv.conf` guaranteed to exist on Bottlerocket only and not on any other supported distros? I second @samjkon's comment.

Reply:
I checked with a Bottlerocket Engineer and he confirmed the `/run/netdog/resolv.conf` will only exist in Bottlerocket.

Reply:
The risk with this approach is that you're coupling the ECS Agent with special behavior which is OS/distro-specific. Should Bottlerocket change this behavior, or if other OSes introduce different behavior, then this pattern suggests that we should update the ECS Agent to accommodate that behavior. Instead is it possible to introduce a configuration value to override the default `resolv.conf` location? By default the option can point to `/etc/resolv.conf`; should you need to override it, it can instead be configured with whatever distro the ECS Agent is included on.